linux/shim/service.go

```diff
@@ -32,23 +31,11 @@ var empty = &google_protobuf.Empty{}
 const RuncRoot = "/run/containerd/runc"
 
 // NewService returns a new shim service that can be used via GRPC
-func NewService(path, namespace, address string) (*Service, error) {
+func NewService(path, namespace string, client publisher) (*Service, error) {
```

Comment:
Should probably use the `events.Publisher` interface, rather than the GRPC interface. Once this PR is in, I can replumb it.